src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,474 @@
+    }
+
+    private static async Task<Solution> MoveStringToResxAndReplaceLiteralAsync(
+        Document document,
+        LiteralExpressionSyntax stringLiteral,
+        TextDocument resxFile,
+        CancellationToken cancellationToken)
+    {
+        var value = stringLiteral.Token.ValueText;
+        var resourceKey = ToDeterministicResourceKey(value);
+
+        var resourceOperation = new ResourceOperation(value, resourceKey);
+        var replacementOperation = new ReplacementOperation(stringLiteral, resourceKey);
+
+        return await ResxResourceManager.UpdateResxAndDocumentAsync(
+            document,
+            resxFile,
+            [resourceOperation],
+            [replacementOperation],
+            cancellationToken).ConfigureAwait(false);
+    }
+
+    private static string ToDeterministicResourceKey(string value, int maxLength = 60)
+    {
+        if (string.IsNullOrWhiteSpace(value))
+            return "emptyString";
+
+        var words = new List<string>();
+        var stringBuilder = new StringBuilder();
+        foreach (var c in value)
+        {
+            if (char.IsLetterOrDigit(c))
+            {
+                stringBuilder.Append(c);
+            }
+            else if (stringBuilder.Length > 0)
+            {
+                words.Add(stringBuilder.ToString().ToLowerInvariant());
+                stringBuilder.Clear();
+            }
+        }
+        if (stringBuilder.Length > 0)
+            words.Add(stringBuilder.ToString().ToLowerInvariant());
+
+        using var _ = PooledStringBuilder.GetInstance(out var keyBuilder);
+        for (var i = 0; i < words.Count; i++)
+        {
+            var word = words[i];
+            if (i == 0)
+            {
+                // Check if adding the first word would exceed maxLength
+                if (word.Length > maxLength)
+                {
+                    keyBuilder.Append(word.Substring(0, maxLength));
+                    break;
+                }
+                keyBuilder.Append(word);
+            }
+            else if (word.Length > 0)
+            {
+                var capitalizedWord = char.ToUpperInvariant(word[0]) + word.Substring(1);
+
+                // Check if adding this word would exceed maxLength
+                if (keyBuilder.Length + capitalizedWord.Length > maxLength)
+                {
+                    // Add as much of the word as possible without exceeding maxLength
+                    var remainingLength = maxLength - keyBuilder.Length;
+                    if (remainingLength > 0)
+                    {
+                        keyBuilder.Append(capitalizedWord.Substring(0, remainingLength));
+                    }
+                    break;
+                }
+
+                keyBuilder.Append(capitalizedWord);
+            }
+        }
+
+        var key = keyBuilder.ToString();
+        if (key.Length == 0)
+            key = "emptyString";
```

Comment:
we should not be getting here.  the analyzer should be validating that we are only offering this on strings we can reasonably create a key for.